File: LeetCode/LeetCode Practice Problems/557- Reverse Words in a String III.cpp
class Solution {
public:
    string reverseWords(string s) {
        s += " ";
        int n = s.length();
        
        string ans = "";
        string cur = "";
        
        for(int i=0;i<n;i++) {
            if(s[i] != ' ') {
                cur = s[i] + cur;
            } else {
                if(ans != "")
                    ans += " ";
                ans += cur;
                cur = "";
            }
        }
        
        return ans;
        
    }
};

File: LeetCode/LeetCode Practice Problems/557-Reverse Words in a String III.java
class Solution {
    public String reverseWords(String s) {
        String ans = "";
        String cur = "";
        for(char c: s.toCharArray()) {
            if(c != ' ') {
                cur = c + cur;
            } else {
                if(ans.length() > 0) {
                    ans = ans + " ";
                }
                ans = ans + cur;
                cur = "";
            }
        }
        
        if(cur.length() > 0) {
            if(ans.length() > 0) {
                ans = ans + " ";
            }
            ans = ans + cur;
        }
        
        return ans;
    }
}
